docs/genai/api/java.md

```diff
@@ -0,0 +1,595 @@
+Set any other search options via the GeneratorParams object as needed using `setSearchOption`.
+
+```java
+GeneratorParams generatorParams = genAI.createGeneratorParams(promptText);
+// .. set additional generator params before calling generate()
+```
+
+Call generate with the GeneratorParams object and an optional listener.
+
+```java
+String fullResponse = genAI.generate(generatorParams, listener);
+```
+
+The listener is used as a callback mechanism so that tokens can be used as they are generated. Create a class that implements the `Consumer<String>` interface and provide an instance of that class as the `listener` argument.
+
+### Constructor
+
+```java
+public SimpleGenAI(String modelPath) throws GenAIException
+```
+
+#### Throws
+
+`GenAIException`- on failure.
+
+### Generate Method
+
+Generate text based on the prompt and settings in GeneratorParams. 
+
+NOTE: This only handles a single sequence of input (i.e. a single prompt which equates to batch size of 1).
+
+```java
+public String generate(GeneratorParams generatorParams, Consumer<String> listener) throws GenAIException
+```
+
+#### Parameters
+
+- `generatorParams`: the prompt and settings to run the model with.
+- `listener`: optional callback for tokens to be provided as they are generated. 
+
+NOTE: Token generation will be blocked until the listener's `accept` method returns.
+
+#### Throws
+
+`GenAIException`- on failure.
+
+#### Returns
+
+The generated text.
+
+#### Example
+
+```java
+SimpleGenAI generator = new SimpleGenAI(modelPath);
+GeneratorParams params = generator.createGeneratorParams("What's 6 times 7?");
+Consumer<String> listener = token -> logger.info("onTokenGenerate: " + token);
+String result = generator.generate(params, listener);
+
+logger.info("Result: " + result);
+```
+
+### createGenerateParams Method
+
+Create the generator parameters and add the prompt text. The user can set other search options via the GeneratorParams object prior to running `generate`.
+
+```java
+public GeneratorParams createGeneratorParams(String prompt) throws GenAIException
+```
+
+#### Parameters
+
+- `prompt`: the prompt text to encode.
+
+#### Throws
+
+`GenAIException`- on failure.
+
+#### Returns
+
+The generator parameters.
+
+## Exception Class
+
+An exception which contains the error message and code produced by the native layer.
+
+### Constructor
+
+```java
+public GenAIException(String message)
+```
+
+#### Example
+
+```java
+catch (GenAIException e) {
+  throw new GenAIException("Token generation loop failed.", e);
+}
+```
+
+## Model class
+
+### Constructor
+
+```java
+Model(String modelPath)
+```
+
+### Create Tokenizer Method
+
+Creates a Tokenizer instance for this model. The model contains the configuration information that determines the tokenizer to use.
+
+```java
+public Tokenizer createTokenizer() throws GenAIException
+```
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails
+
+#### Returns
+
+`onnxruntime_genai.Model`
```

Comment:
This should return a Tokenizer not a Model

Reply:
fixed.